feat: 支持限定 vm 集群查询 --story:124795901 --story=124795901

Zhaoyikaiii opened the pull request.

Zhaoyikaiii committed:
feat: 支持限定 vm 集群查询

Zhaoyikaiii committed:
chore: 添加vm-force-storage-name trace

Zhaoyikaiii committed:
fix: 补充缺少的强制使用配置的storageName位置

shamcleren committed:
Merge remote-tracking branch 'upstream/master' into feat/vm_cluster_qualified_query/#1010158081124795901

# Conflicts:
#	pkg/unify-query/unify-query.yaml

shamcleren commented on pkg/unify-query/tsdb/victoriaMetrics/instance.go:
```diff
@@ -78,6 +79,8 @@ type Instance struct {
 	timeout time.Duration
 	curl    curl.Curl
+
+	forceStorageName string
```
还有其他的几个 ClusterName 都没有覆盖，建议把这个整体封装成一个方法，后面统一调用这个方法

Zhaoyikaiii replied:
已修改

Zhaoyikaiii committed:
refactor: 修改vm instance设置clusterName

- 使用i.getClusterName(clusterNames ...) 获取集群名字
- i.forceStorageName优先级最高

Zhaoyikaiii committed:
chore: 修改vm强存储命名配置

- victoria_metrics.force_query_storage_name -> query_router.force_query_storage_name

Zhaoyikaiii committed:
chore: add debug vm force storage name log

Zhaoyikaiii committed:
fix: viper没有获取到配置

shamcleren commented on pkg/unify-query/service/tsdb/hook.go:
```diff
@@ -81,6 +83,8 @@ func initConfig() {
 	VmInfluxCompatible = viper.GetBool(VmInfluxCompatibleConfigPath)
 	VmUseNativeOr = viper.GetBool(VmUseNativeOrConfigPath)
+	VmForceQueryStorageName = viper.GetString(VmForceQueryStorageNameConfigPath)
+	log.Printf("force query storage name: %s", VmForceQueryStorageName)
```
这里要删掉

Zhaoyikaiii replied:
已移除。之前是用来进行调试

shamcleren committed:
Merge branch 'master' into feat/vm_cluster_qualified_query/#1010158081124795901

Zhaoyikaiii committed:
chore: 调整调试vmForceStorageNameConfig日志

Zhaoyikaiii committed:
chore: 移除调试vm强制storageName日志

Zhaoyikaiii committed:
Merge branch 'master' into feat/vm_cluster_qualified_query/#1010158081124795901

shamcleren committed:
feat: 支持限定vm集群查询 #1010158081124795901

shamcleren committed:
feat: 支持限定vm集群查询 #1010158081124795901

shamcleren commented on pkg/unify-query/service/tsdb/hook.go:
```diff
@@ -47,6 +47,7 @@ func setDefaultConfig() {
 	// vm 支持 influxdb 的查询配置
 	viper.SetDefault(VmInfluxCompatibleConfigPath, true)
 	viper.SetDefault(VmUseNativeOrConfigPath, true)
+	viper.SetDefault(VmForceQueryStorageNameConfigPath, "")
```
为空的话，就不需要配置默认值了

shamcleren committed:
feat: 支持限定vm集群查询 #1010158081124795901
